registry/storage/shares.go

```diff
@@ -364,35 +369,17 @@ func (s *sharesStorage) UpdateValidatorsMetadata(data map[spectypes.ValidatorPK]
 		}
 	}()
 
-	saveShares := func(sshares []*types.SSVShare) error {
-		s.mu.Lock()
-		defer s.mu.Unlock()
-		if err := s.unsafeSave(nil, sshares...); err != nil {
-			return err
-		}
-		return nil
-	}
-
-	// split into chunks to avoid holding the lock for too long
-	chunkSize := 1000
-	for i := 0; i < len(shares); i += chunkSize {
-		end := i + chunkSize
-		if end > len(shares) {
-			end = len(shares)
-		}
-		if err := saveShares(shares[i:end]); err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return s.Save(nil, shares...)
 }
 
 // Drop deletes all shares.
 func (s *sharesStorage) Drop() error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
```

Comment:
Can maybe lock `mu` just for these lines (and then also have to `dbmu.Unlock()` before going for `mu.Lock`):
```go
s.shares = make(map[string]*types.SSVShare)
s.validatorStore.handleDrop()
```
Essentially, same optimization you've made for `sharesStorage.Save` above, seems like this is applicable to other func(s) too, eg. `Delete` (and maybe others).

Reply:
added